contracts/PatriciaTree/PatriciaTreeProofs.sol

```diff
@@ -0,0 +1,41 @@
+pragma solidity ^0.4.16;
+pragma experimental "v0.5.0";
+pragma experimental "ABIEncoderV2";
+
+import {Data} from "./Data.sol";
+import {Bits} from "./Bits.sol";
+import "./IPatriciaTree.sol";
+import "./PatriciaTreeProofs.sol";
+
+/// @title Patricia tree implementation
+/// @notice More info at: https://github.com/chriseth/patricia-trie
+contract PatriciaTreeProofs {
+  using Bits for uint;
+  using Data for Data.Edge;
+  using Data for Data.Label;
+
+  function verifyProof(bytes32 rootHash, bytes key, bytes value, uint branchMask, bytes32[] siblings) public view returns (bool) {
```

Comment:
How about removing this function altogether? The pattern of passing in the validation value (rootHash) here feels wrong. Also this function never returns `false` which makes its return type like `true-or-throw` instead of `bool`. Can we rather push this 1 check to wherever this is referenced which will also allow us to pass in a different `require` fail error message back?